internal/controllers/metrics.go

```diff
@@ -35,6 +35,31 @@ func (c ApiController) SaveMetricsHandler(w http.ResponseWriter, r *http.Request
 
 	requestContext := r.Context().Value(auth.RequestContextKey).(*auth.RequestContext)
 
+	// check if node with provided ID exists
+	node, err := c.repositories.NodeRepo.FindByID(requestContext.NodeId)
```

Comment:
It would be more elegenat to have another metrics repo methods that would save and return if it is first and then we can have one if to add node to active if it is to avoid so much ifs

Reply:
I agree, will refactor :+1: